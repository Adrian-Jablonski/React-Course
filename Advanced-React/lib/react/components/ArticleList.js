import React from 'react';
import Article from './Article';
import PropTypes from 'prop-types';

const ArticleList = ({articles, articleActions}) => {
	return (
		<div>
			{Object.values(articles).map((article, index) => {
				return (
					<Article 
						key={index}
						article={article}
						actions={articleActions}
					/>
				);
			})}
		</div>
	);
};

export default ArticleList;

ArticleList.propTypes = {
	articles: PropTypes.object.isRequired,
	articleActions: PropTypes.object.isRequired
};