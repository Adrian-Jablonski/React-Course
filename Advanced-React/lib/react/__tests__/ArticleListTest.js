import React from 'react';
import ArticleList from '../components/ArticleList';

import renderer from 'react-test-renderer';

describe('ArticleList', () => {

	const testProps = {
		articles: {
			a: {id: 'a'},
			b: {id: 'b'}
		},
		articleActions: {
			lookupAuthor: jest.fn(() => ({})),
		}
	}

	it('renders correctly', () => {
		const tree = renderer.create(
			<ArticleList 
				{...testProps}
			/>
		).toJSON();

		expect(tree.children.length).toBe(2);

		expect(tree).toMatchSnapshot(); // Takes a snapshot the first time the test is run. And compares the snapshot taken to each following snapshot. The snapshot is saved in the __snapshots__ folder
		// Making any changes to component will cause the test to fail, but you could inspect the changes in the console and press u to update the snapshot
	});


});