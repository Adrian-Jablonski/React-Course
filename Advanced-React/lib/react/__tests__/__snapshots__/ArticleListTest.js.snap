// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <p
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <p
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Invalid Date
    </p>
    <p
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
       
    </p>
    <p
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <p
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    />
    <p
      style={
        Object {
          "color": "#888",
          "fontSize": "0.8em",
        }
      }
    >
      Invalid Date
    </p>
    <p
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
       
    </p>
    <p
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    />
  </div>
</div>
`;
